src/olympia/amo/middleware.py

```diff
@@ -95,7 +99,8 @@ class AuthenticationMiddlewareWithoutAPI(AuthenticationMiddleware):
     own authentication mechanism.
     """
     def process_request(self, request):
-        if request.path.startswith('/api/'):
+        if (request.path.startswith('/api/') and
+                not auth_view.match(request.path)):
```

Comment:
Prepend your regexp with a  `^` or even just use `not request.path.startswith(...)`